ci/jest-reporter.js

```diff
@@ -1,47 +1,98 @@
+const { generateDiscordCloudwatchLogFile, listECSTasks, sendDiscordNotification } = require('./helpers')
+
 const { BaseReporter } = require('@jest/reporters')
 const child_process = require('child_process')
 
+async function main_task() {
+
+  taskArns = await listECSTasks()  // like taskArns = ['arn:aws:ecs:us-east-2:967314784947:task/ceramic-dev-tests/2466935a544f47ec9a1c3d8add235c84']
+
+  if (taskArns.length > 1) {
+    console.warn('WARN: NEEDS INVESTIGATION, more than one task running !!!')
+  } else if (taskArns.length === 1) {
+    console.log('INFO: OK, only one running task found (assumed to be self)')
+  } else {
+    console.warn('WARN: NEEDS INVESTIGATION, no task found running !!!')
+  }
+
+  console.log("INFO: taskArns:=", taskArns)
+}
+
+
+
 class MyCustomReporter extends BaseReporter {
   constructor(globalConfig, options) {
     super(globalConfig, options)
     this._globalConfig = globalConfig
     this._options = options
     this.runId = process.env.RUN_ID
+    this.taskArns = []
+    this.LogFiles = []
   }
 
   onRunStart(results, options) {
-    const message = buildDiscordStart(results, this.runId)
-    const data = { embeds: message, username: 'jest-reporter' }
-    const out = child_process.execSync(
-      `curl -X POST \
-        -H "Content-Type: application/json" \
-        -d '${JSON.stringify(data)}' \
-        ${process.env.DISCORD_WEBHOOK_URL}`
-    )
-    console.log(out.toString())
+    main_task().then(() => {
+      console.log('Done !!!')
+      this.taskArns = taskArns
+      this.LogFiles = generateDiscordCloudwatchLogFile(taskArns)
+      //ACTUAL// this.testPassUrl = process.env.DISCORD_WEBHOOK_URL_TEST_RESULTS
+      //ACTUAL// this.testFailUrl = process.env.DISCORD_WEBHOOK_URL_TEST_FAILURES
+      this.testPassUrl = 'https://discordapp.com/api/webhooks/877880441254842388/OnoMzNAHIzWawEymW7Cp_2WNOCjggtoJyjFnMOTx_ut43yFlp4RRSQKd8W6vGjgQiL9O' //TEST//
+      this.testFailUrl = 'https://discordapp.com/api/webhooks/879318039064100894/fcAM5HypKLuwWIIcYEoOzVEI6eLIv6Mah3Af991NbE75zfMZPlfVbT2S_dde10pUt6xn' //TEST//
+
+      const message = buildDiscordStart(results, this.runId, this.LogFiles)
+      const data = { embeds: message, username: 'jest-reporter' }
```

Comment:
I guess we can pull out the `username` value into a constant

Reply:
Yeah, calling it userName